### DISCUSSION TITLE
Create a Color Blind Palette by default

### DISCUSSION THREAD

**Discussion by nacengineer** [tailwindlabs/tailwindcss#8428] [Ideas]
It would be really nice to have a color blind palette by default vs having to layer one into the design. It's something that is usually an afterthought to most designers IME.  In the past I have used this I use this article as a guide 

http://mkweb.bcgsc.ca/colorblind/

They have a [few color palettes linked here](http://mkweb.bcgsc.ca/colorblind/palettes.mhtml#page-container), e.g. 8, 12, 15, 24 color palettes represented that do well for all CB types. 

Thanks for the consideration. 😀

**Comment by vinksz** [comment]
I have a need for such thing atm too.  There's a PostCSS plugin for example ->

```
var fs = require("fs")
var postcss = require("postcss")
var colorblindPlugin = require("postcss-colorblind");

var css = fs.readFileSync("input.css", "utf8");

postcss()
  .use(colorblindPlugin({method:'achromatopsia'}))
  .process(css)
  .then(function(result) {
    fs.writeFileSync('output.css', result.css);
  });
```

Ive never tried to extend tailwind or make tailwind plugin before...  need to take look into it :)

**Reply by vinksz** [reply]
Noone probably cares because there's % generally, not many colorblind people, unless you care about minorities.
There are an estimated 300 million people in the world with color vision deficiency. 1 in 12 men are color blind (8%). 1 in 200 women are color blind (0.5%).

Would make sense, that Browsers themselves support colorblinds and auto-convert colors.

**Reply by nacengineer** [reply]
IME no one cares because I've found color blind people just accept things as the norm or actually don't realize there is a difference, until you demonstrate it. There are some that are too embarrassed to admit to being color blind, maybe this has to do with it largely affecting men? Just a wild ass guess as I'm not sure. However I think most of the color blind people I have designed for just didn't realize the difference a CB palette makes. One of the most egregious default CB non friendly apps we use is Confluence and you'd think they'd know better. I designed a Chrome extension for a coworker that flipped our confluence wiki to CB friendly colors and he was shocked when it flipped things to links that he had always thought were static text.  

However IMO it seems like a minimum viable disability feature to have for a CSS framework. Most games have CB mode, quite a few chat clients have CB modes, e.g. slack. I would normally say yes it's fine to push that into the browser for a product as but as this is a framework for visual design, I'm of the opinion that something "hand designed" like TailwindCSS could easily bridge the gap. 😀
